tchannel-core/src/main/java/com/uber/tchannel/api/SubChannel.java

```diff
@@ -78,6 +78,12 @@ public SubChannel(String service, @NotNull TChannel topChannel) {
     }
 
     public SubChannel(String service, @NotNull TChannel topChannel, Connection.Direction preferredDirection) {
+        if (topChannel == null) {
+            throw new NullPointerException("`topChannel` cannot be null");
+        }
+        if (preferredDirection == null) {
```

Comment:
there's no evidence this must be non-null (though it might've been the original intent).